presto-main/src/main/java/com/facebook/presto/metadata/PolymorphicScalarFunction.java

```diff
@@ -148,16 +163,31 @@ private boolean predicateIsTrue(MethodsGroup methodsGroup, SpecializeContext con
         return methodsGroup.getExtraParametersFunction().map(function -> function.apply(context)).orElse(emptyList());
     }
 
+    private int getNullFlagsCount()
+    {
+        int count = 0;
+        for (Boolean flag : nullFlags) {
+            if (flag) {
+                count++;
+            }
+        }
+        return count;
+    }
+
     private MethodHandle applyExtraParameters(Method matchingMethod, List<Object> extraParameters)
     {
         Signature signature = getSignature();
-        int expectedNumberOfArguments = signature.getArgumentTypes().size() + extraParameters.size();
+        int argumentsWithNullableFlagCount = signature.getArgumentTypes().size() + getNullFlagsCount();
```

Comment:
If I rename `argumentsWithNullableFlagCount` it looks better with functions. It reads: let's take parameters and add null flags to it. What makes it less readable?